test/test_scandir.py

```diff
@@ -250,6 +250,32 @@ def test_unicode(self):
         self.assertEqual(entry.name, 'file1.txt')
         self.assertEqual(entry.path, os.path.join(path, 'file1.txt'))
 
+    def test_walk_unicode_handling(self):
+        encoding = sys.getfilesystemencoding()
+        subdir = os.path.join(TEST_PATH, 'walk').encode(encoding)
```

Comment:
I don't understand why you need 3 levels of directories: TEST_PATH/walk/dada? Why not one level: TEST_PATH?